### DISCUSSION TITLE
Editor suggesting code fixes that don't need to occur (csharp)

### DISCUSSION THREAD

**Discussion by joro550** [helix-editor/helix#6233] [General]
Hey there,

I'm not really sure if this is a omnisharp thing (all though it doesn't seem to happen in Vs Code) but when I load my project directory up in helix and navigate to a controller it seems not to be registering everything correctly

So for example:

![image](https://user-images.githubusercontent.com/1705979/221796466-b68cc3d6-23a4-4641-b30f-5d5850247bec.png)

This tells me that System.Security.Cryptography is not needed in my file (using it for Md5) and simultaneous tells me that MD5 class doesn't exist in the current context

When I run `hx --health c-sharp` this is the output:

```
Configured language server: OmniSharp
Binary for language server: C:\Users\{User}\.omnisharp\OmniSharp.exe
Configured debug adapter: netcoredbg
Binary for debug adapter: Not found in $PATH
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✘
```

But seeing as it is a language specific thing I'm unsure if this is omni sharp telling the editor this is incorrect and not the editor itself

**Comment by joro550** [comment]
Oh interestingly if I open the folder where the csproj file resides I don't have these issues, usually I work where I open the folder where the sln is so:
```
/project
/project/.csproj
/project-tests
/protect-tests/.csproj
project.sln
```

I usually open it in `/` directory but that's when I get the errors, but if I open `/project` then I don't get the issue, strange

**Comment by askreet** [comment]
I don't know C# well, but the editor only starts LSPs and talks to them, it doesn't know anything about C#. That said, it does do some logic to determine what a project root is and I think that scopes where/how the LSP runs:

https://github.com/helix-editor/helix/blob/master/languages.toml#L254

For some languages when you have nested projects it's a matter of the LSP properly understanding where to find the root project file (project.sln?) and wiring that all together. I'd suggest starting with the LSP project for troubleshooting and then possibly Helix's project detection needs tweaking for C#.
